src/components/SQForm/SQFormReadOnly.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import Grid from '@material-ui/core/Grid';
+import {Formik, Form} from 'formik';
+import SQFormReadOnlyField from './SQFormReadOnlyField';
+import SQFormMaskedReadOnlyField from './SQFormMaskedReadOnlyField';
+import {noop} from '../../utils';
+import type {CSSProperties} from 'react';
+import type {GridProps} from '@material-ui/core';
+import type {FormikValues} from 'formik';
+import type {SQFormMaskedReadOnlyFieldProps} from './SQFormMaskedReadOnlyField';
+
+export interface SQFormReadOnlyProps<Values extends FormikValues> {
+  /** Form Input(s) */
+  readOnlyFields: SQFormMaskedReadOnlyFieldProps[];
+  /** Form Entity Object */
+  initialValues: Values;
+  /** Any prop from https://material-ui.com/api/grid */
+  muiGridProps?: GridProps;
+}
+
+const noBottomMargin: CSSProperties = {
+  marginBottom: 0,
+};
+const noBottomMarginStyle = {style: noBottomMargin};
+
+function SQFormReadOnly<Values extends FormikValues>({
+  readOnlyFields,
+  initialValues,
+  muiGridProps = {},
+}: SQFormReadOnlyProps<Values>): JSX.Element {
+  return (
+    <Formik<Values> initialValues={initialValues} onSubmit={noop}>
+      {(_props) => {
+        return (
+          <Form>
+            <Grid
+              {...muiGridProps}
+              container
+              spacing={muiGridProps.spacing ?? 2}
+            >
+              {readOnlyFields.map((readOnlyField, index) => {
+                const props = {
+                  key: index,
```

Comment:
In general we should avoid using array indexes alone for element keys. Perhaps we could use the name of the field passed in? Another option would be to require the consumer to pass it in by typing `readOnlyField` as `SQFormMaskedReadOnlyFieldProps[] & { key: string}`

Reply:
We've tried using `name` for the `key` in the past but it quickly became a problem as name from our data was frequently not unique. 

In general we should use array indexes as a last resort. This is a read only form so we can't manipulate, delete, shift around the order of the fields so there shouldn't be any ill effect of using `index`.

If we require the type in the TS `type` the consumer is likely just going to use the array index. A complex key doesn't move the bar forward much either, as `name` and `label` can both be repeated as we've seen on the Admin views project.

Using a date time stamp for each and every key is really expensive.

In this case, I think `index` as the last resort is viable

https://reactjs.org/docs/lists-and-keys.html#keys

> We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state. Check out Robin Pokorny’s article for an [in-depth explanation on the negative impacts of using an index as a key](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318). If you choose not to assign an explicit key to list items then React will default to using indexes as keys.

From the above link in the React docs:

"Update: Exception from the rule
Many people asked if they always, always have to generate ids. Others have suggested use cases when using the index as a key seems justifiable.
It is true that sometimes generating new ids is redundant and may be avoided. For example translation of license terms or list of contributors.
To help you decide, I put together three conditions which these examples have in common:
the list and items are static–they are not computed and do not change;
the items in the list have no ids;
the list is never reordered or filtered.
When all of them are met, you may safely use the index as a key."

Reply:
I'd argue that as a library, we can't guarantee that all of those conditions will be met. A consumer can definitively make that assumption for their use case, however.

Reply:
To provide a simple example of when one of these conditions owuld not be met, and a non-contrived example, would be when loading the read-only data from an endpoint. Initially the value of the fields are likely an empty string but then when the data is loaded it's changed.

Similarly, a form that displays read-only data based on the selection of another element in the UI.

Reply:
> To provide a simple example of when one of these conditions owuld not be met, and a non-contrived example, would be when loading the read-only data from an endpoint. Initially the value of the fields are likely an empty string but then when the data is loaded it's changed.
> 
> Similarly, a form that displays read-only data based on the selection of another element in the UI.

I'm happy to add `index` to the `type` so that the consumer can make the choice of whether or not to use `index` rather than the library.

The 2 real world examples shouldn't be affected by using `index` as the key. If the value of the props need to change in order to reflect new values in the form then we set `enableReinitialize` within the formik form.

Reply:
I think a happy medium here would be to allow the consumer to pass an optional key and have consumer provided keys take precedence over whatever the library decides to use. 
You're likely right in that we'd never see an issue but the ones that would crop out would be quite the doozies to try to debug. This way consumers could at least alleviate one potential issue. Plus, if they need to identify the elements in the DOM for whatever reason they can use their own keys instead.

Reply:
Developer is now able to specify React key's
cc: @20BBrown14